### PR TITLE
Feature/dhclient configs v6

### PR DESCRIPTION
**Pull request by metelik**
1. Shifting the purpose of dhclientv4_conf.ex into serving both purposes Ipv4 and Ipv6 - because the code would be in like 90% common for both.
2. Launching the DhclientConf module in two flavours Ipv4 and Ipv6 in Nerves.Network application

### CODE REVIEW

File: lib/nerves_network/application.ex [hunk 1]
```diff
@@ -9,14 +9,19 @@ defmodule Nerves.Network.Application do
...
 
     children = [
       supervisor(Registry, [:duplicate, Nerves.Dhclientv4], id: Nerves.Dhclientv4),
       supervisor(Registry, [:duplicate, Nerves.Dhclient], id: Nerves.Dhclient),
       worker(Nerves.Network.Resolvconf, [resolvconf_file, [name: Nerves.Network.Resolvconf]]),
-      worker(Nerves.Network.Dhclientv4Conf, [dhclientv4_config_file, [name: Nerves.Network.Dhclientv4Conf]]),
+      Supervisor.child_spec({Nerves.Network.DhclientConf, [dhclientv4_config_file, [name: Nerves.Network.DhclientConf.Ipv4]]}, id: Nerves.Network.DhclientConf.Ipv4),
+      Supervisor.child_spec({Nerves.Network.DhclientConf, [dhclientv6_config_file, [name: Nerves.Network.DhclientConf.Ipv6]]}, id: Nerves.Network.DhclientConf.Ipv6),
+      #worker(Nerves.Network.DhclientConf, [dhclientv4_config_file, [name: Nerves.Network.DhclientConf.Ipv4]]),
+      #worker(Nerves.Network.DhclientConf, [dhclientv6_config_file, [name: Nerves.Network.DhclientConf.Ipv6]]),
```
Comment (michaelkschmidt) [inline lib/nerves_network/application.ex]: If these aren't needed its best to delete

File: lib/nerves_network/dhclient_conf.ex [hunk 2]
```diff
@@ -109,24 +111,25 @@ defmodule Nerves.Network.Dhclientv4Conf do
   Start the resolv.conf manager.
   """
   @spec start_link(Path.t, GenServer.options) :: GenServer.on_start
-  def start_link(dhclient_conf_path \\ @dhclient_conf_path, opts \\ []) do
-    GenServer.start_link(__MODULE__, dhclient_conf_path, opts)
+  def start_link([dhclient_conf_path, [name: server_name]], opts \\ []) do
+    IO.puts "#{__MODULE__}: start_link: server_name = #{inspect server_name} dhclient_conf_path = #{inspect dhclient_conf_path} opts = #{inspect opts}"
```
Comment (michaelkschmidt) [inline lib/nerves_network/dhclient_conf.ex]: Convert this to a log statement or remove

**Review by michaelkschmidt — APPROVED** [review]
Looks good.  Just a couple cleanup items

File: lib/nerves_network/application.ex (hunk 1 above)
Comment (metelik) [reply lib/nerves_network/application.ex]: done

File: lib/nerves_network/dhclient_conf.ex (hunk 2 above)
Comment (metelik) [reply lib/nerves_network/dhclient_conf.ex]: done

**Review by bjmurphySE — APPROVED** [review]
This will need a good test

**Pull request merged**